### ISSUE TITLE
configure_file cannot output file in subfolders of the build directory

### ISSUE THREAD

**Issue by amyspark** [mesonbuild/meson#11576]
**Describe the bug**

Sometimes, it is necessary to rename a file to make it fit into an external script or build tree. For instance, when building a wrap of a library that installs headers to assorted subfolders of `includedir`. For this, `configure_file` is the obvious choice pre-`fs`module.

However, while it allows copying a file **from** a subfolder of the current source directory, it does not allow copying **to** a subfolder. This breaks building consumers agnostically, requiring custom checks to change the include path on whether it's an `internal` dep or not.

**To Reproduce**

The following meson.build reproduces the copy error:

```meson
project('test', 'c')

srcs = []

srcs += configure_file('main.c', 'src/main.c', copy: true)

exe = executable('test', srcs)
```

main.c:

```c
int main() { return 0; }
```

**Expected behavior**

The copy should succeed. Instead `configure_file keyword "output" Output 'src/main.c'  must not contain a path segment.` is triggered.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer) **Native**
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.) **Windows 10 21H2**
* what Python version are you using e.g. 3.8.0 **3.10.7**
* what `meson --version` **1.0.1**
* what `ninja --version` if it's a Ninja build **1.10.2.git.kitware.jobserver-1**

**Comment by eli-schwartz** [comment]
> However, while it allows copying a file **from** a subfolder of the current source directory, it does not allow copying **to** a subfolder. This breaks building consumers agnostically, requiring custom checks to change the include path on whether it's an `internal` dep or not.

This is false, you simply define the configure_file or fs.copyfile inside `subfolder/meson.build`. The real question is not whether you can output targets to subfolders, but whether you should be obliged to create a meson.build file in that subfolder in order to do it.

It is, admittedly, an opinionated design decision of meson. But it's a consistent design decision, that applies to all target types -- custom_target, configure_file, executable, shared_library, etc. Which means the title of this ticket is incorrect, as it has nothing to do with configure_file per se.
